### ISSUE TITLE
Testing in dp mode uses only one of the GPUs

### ISSUE THREAD

**Issue by Ir1d** [PyTorchLightning/pytorch-lightning#1213]
### To Reproduce

Run a test without training

![image](https://user-images.githubusercontent.com/10709657/77311956-1706ac00-6d3c-11ea-89fe-3cf2156babe2.png)

#### Code sample

Modified from the conference-seed repo

```python
trainer = Trainer(
            gpus="-1",
            distributed_backend='dp',
        )
trainer.test(model)
```

### Environment

 - pl version: 0.6.0
 - PyTorch Version (e.g., 1.0): 1.2
 - OS (e.g., Linux): Ubuntu
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration:
 - Any other relevant information:

**Comment by williamFalcon** [comment]
ummm yeah, that's a bug. it should run via dp. @Ir1d want to submit a PR?
@PyTorchLightning/core-contributors any one else experience this?

**Comment by Ir1d** [comment]
@williamFalcon I tried wrapping the model in `LightningDataParallel` like the training loop did, but it tells me that LightningDataParallel object doen't have a `test_step`. How do I debug this?

**Comment by williamFalcon** [comment]
you wouldn’t wrap it yourself ever haha. 
the trainer does the wrapping for you. 

the trainer needs to be modified to run the test on the correct method when done this way

**Comment by Ir1d** [comment]
I was trying to wrap it in `evaluate` in `pytorch_lightning/trainer/evaluation_loop.py` . Do you have any idea where to wrap this func?

**Comment by Ir1d** [comment]
Anyway, we've find one possible workround here:
After defining a torch model, and before sending it into PL model, wrap it with nn.dataparallel.

**Comment by williamFalcon** [comment]
evaluate is private... you're not meant to call it directly.

call .test()

**Comment by williamFalcon** [comment]
lightning does the wrapping by itself... 

the fact that this doesn't work, is a bug.

```python
model = MyLightningModule.load_from_checkpoint(...)
trainer = Trainer(
            gpus="-1",
            distributed_backend='dp',
        )
trainer.test(model)
```

The bug needs to be addressed correctly.

It's weird because we have tests for this... double check that this is really not working for you.

**Comment by Borda** [comment]
> evaluate is private... you're not meant to call it directly.
> call .test()

so let's rename it starting with _ to be clear that it is private from it name

**Comment by Ir1d** [comment]
I was calling .test and its not working

**Comment by Borda** [comment]
@neggert may you have look at this multi GPU issue?

**Comment by edenlightning** [comment]
@neggert ping :)

**Comment by williamFalcon** [comment]
looking at this with next sprint

**Comment by williamFalcon** [comment]
fixed!
